experimental/examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/ImageViewer.common.kt

```diff
@@ -1,35 +1,71 @@
 package example.imageviewer
 
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.fadeIn
+import androidx.compose.animation.fadeOut
+import androidx.compose.animation.slideInHorizontally
+import androidx.compose.animation.slideOutHorizontally
 import androidx.compose.foundation.layout.fillMaxSize
-import androidx.compose.material.Surface
+import androidx.compose.material3.Surface
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.remember
 import androidx.compose.ui.Modifier
-import example.imageviewer.model.*
+import example.imageviewer.model.GalleryScreenState
+import example.imageviewer.model.ScreenState
+import example.imageviewer.model.bigUrl
 import example.imageviewer.view.FullscreenImage
 import example.imageviewer.view.MainScreen
+import kotlinx.coroutines.flow.Flow
+
+enum class ExternalImageViewerEvent {
+    Foward,
+    Back
+}
 
 @Composable
-internal fun ImageViewerCommon(state: MutableState<State>, dependencies: Dependencies) {
-    state.refresh(dependencies)
+internal fun ImageViewerCommon(
+    dependencies: Dependencies,
+    externalEvents: Flow<ExternalImageViewerEvent>? = null
```

Comment:
```suggestion
    externalEvents: Flow<ExternalImageViewerEvent> = emptyFlow()
```
We can use default empty flow to simplify nullable checks.
In this case also need to add import ```import kotlinx.coroutines.flow.emptyFlow```

But, nullable is also good enough.

Reply:
I like this idea! Replaced with `emptyFlow()`, and imported as suggested.!